assets/app/scripts/directives/metrics.js

```diff
@@ -0,0 +1,257 @@
+          var sparklineConfig,
+              container = scope.options.selectedContainer,
+              convertUnitsFn = convertUnitsFnByMetric[metric.id];
+          var chartData = scope.chartDataByMetric[metric.id] = {
+            xData: ['dates'],
+            yData: [metric.units]
+          };
+
+          // The patternfly charts change the config objects we pass on
+          // initialization, so just update what's there rather than recreating
+          // the object if it exists.
+          if (!scope.utilizationConfigByMetric[metric.id]) {
+            scope.utilizationConfigByMetric[metric.id] = {
+              units: metric.units
+            };
+          }
+
+          if (!scope.donutConfigByMetric[metric.id]) {
+            scope.donutConfigByMetric[metric.id] = {
+              units: metric.units,
+              chartId: metric.chartId
+            };
+          }
+
+          if (!scope.sparklineConfigByMetric[metric.id]) {
+            // Make a copy of the base sparkline config to edit it.
+            scope.sparklineConfigByMetric[metric.id] = angular.copy(baseSparklineConfig);
+          }
+          sparklineConfig = scope.sparklineConfigByMetric[metric.id];
+          sparklineConfig.chartId = metric.chartId;
+          sparklineConfig.units = metric.units;
+          sparklineConfig.axis.y.label = metric.units;
+
+          // If we're showing data from another day, add the abbreviated day name
+          // to the time format.
+          var start = moment().subtract(scope.options.timeRange.value, 'minutes');
+          if (start.isSame(moment(), 'day')) {
+            sparklineConfig.axis.x.tick.format = '%H:%M';
+          } else {
+            sparklineConfig.axis.x.tick.format = '%a %H:%M';
+          }
+
+          var limit;
+          switch (metric.id) {
+          case 'memory/usage':
+            var memLimit = getMemoryLimit(container);
+            if (memLimit) {
+              // Convert to MB. usageValueFilter returns bytes.
+              limit = bytesToMB(usageValueFilter(memLimit));
+            }
+            break;
+          case 'cpu/usage':
+            var cpuLimit = getCPULimit(container);
+            if (cpuLimit) {
+              // Remove the 'm' unit at the end.
+              limit = parseInt(cpuLimit, 10);
+            }
+            break;
+          }
+
+          if (limit) {
+            chartData.total = limit;
+            // The utilization sparkline chart is much shorter. Use only a couple ticks.
+            sparklineConfig.axis.y.tick.count = 2;
+          }
+
+          if (data.length) {
+            var lastDataPoint = data[data.length - 1].value;
+            if (isNaN(lastDataPoint)) {
+              lastDataPoint = 0;
+            }
+            var converted;
+            if (convertUnitsFn) {
+              converted = convertUnitsFn(lastDataPoint);
+            }
+            chartData.used = d3.round(converted || lastDataPoint, 1);
```

Comment:
how often is this getting updated?  does this whole function run every time we get a new data point back?